src/main/kotlin/com/yapp/giljob/infra/s3/application/S3Service.kt

```diff
@@ -0,0 +1,37 @@
+package com.yapp.giljob.infra.s3.application
+
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.yapp.giljob.global.error.ErrorCode
+import com.yapp.giljob.global.error.exception.BusinessException
+import com.yapp.giljob.infra.s3.dto.responsne.S3UploadResponseDto
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.multipart.MultipartFile
+import java.util.*
+
+@Service
+class S3Service @Autowired constructor(
+    private var amazonS3: AmazonS3
+){
+    @Value("\${cloud.aws.s3.bucket}")
+    private lateinit var bucketName: String
+
+    fun fileUplaod(file: MultipartFile): S3UploadResponseDto {
+        val fileName: String
+
+        try {
+            fileName = UUID.randomUUID().toString() + file!!.originalFilename
+            val metadata = ObjectMetadata()
+            metadata.contentLength = file!!.size
+            amazonS3.putObject(bucketName, fileName, file!!.inputStream, metadata)
```

Comment:
```suggestion
            val metadata = ObjectMetadata().apply {
              contentLength = file!!.size
            }
            fileName = UUID.randomUUID().toString() + file!!.originalFilename
            amazonS3.putObject(bucketName, fileName, file!!.inputStream, metadata)
```
1. 초기화 시 apply 로 parameter 주입이 가능합니다.
2. (제가 서버를 잘 몰라서...) file 의 non-null 처리를 안할 수는 없을까요? null to non-null 로 캐스팅하는 로직이 순리에 맞지는 않은 것 같습니다. 🤔 file의 nullable 체크를 먼저 진행한 후에 try catch 로 넘어가는게 어떨까요? @s2moon98

Reply:
1. 오 apply 이번에 처음 알았어요!! 적용해보겠습니다
2. 맞는 거 같아요! 제가 기존 자바 코드 방식을 적용하다보니 저렇게 코드를 짠 것 같네요! 수정해볼게용